File: go/isogram/isogram.go
package isogram

import "strings"

// IsIsogram tests whether the word is an isogram.
func IsIsogram(word string) bool {
  charactersSeen := make(map[rune]int)
  const seen = 1

  for _, character := range strings.ToLower(word) {
    if len(word) == 0 || character == ' ' || character == '-' {
      continue
    }

    if _, ok := charactersSeen[character]; ok {
      return false
    }
    charactersSeen[character] = seen
  }

  return true
}

File: python/isogram/isogram.py
import re

def is_isogram(string):
    characters_seen = {}

    if string == '':
        return True

    for character in string.lower():
        if re.match(r'[a-zA-z]+', character):
            if character in characters_seen:
                return False
            else:
                characters_seen[character] = True

    return True
